### PR TITLE
Improve release docs

### PR DESCRIPTION
**Pull request by piyush-garg**
This will add detailed information about some
steps which needs to be performed for releasing
the CLI

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [ ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [ ] Regenerate the manpages and docs with `make docs` and `make man` if needed.
- [ ] Run the code checkers with `make check`
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/cli/blob/master/CONTRIBUTING.md)
for more details._

### CODE REVIEW

File: RELEASE_PROCESS.md [hunk 1]
```diff
@@ -39,9 +39,19 @@ dnf makecache
 dnf upgrade tektoncd-cli
 ```
 
-- Make an update for [homebrew-core](https://github.com/Homebrew/homebrew-core/blob/master/Formula/tektoncd-cli.rb) for tektoncd-cli formula, and make sure you get the proper `sha256sum` and update it like this:
-
-  https://github.com/Homebrew/homebrew-core/pull/46492
+- Make an update for [homebrew-core](https://github.com/Homebrew/homebrew-core/blob/master/Formula/tektoncd-cli.rb) for tektoncd-cli formula.
```
Comment (chmouel) [inline RELEASE_PROCESS.md]: s/for/on/

File: RELEASE_PROCESS.md [hunk 2]
```diff
@@ -39,9 +39,19 @@ dnf makecache
...
+
+  * Make sure you get the proper `sha256sum` You need to download the `Source Code (tar.gz)` from the release and hit below command for the respective release
+  
+    ```shell script
+    openssl dgst --sha256 cli-0.7.1.tar.gz
```
Comment (chmouel) [inline RELEASE_PROCESS.md]: I guess openssl works but so should be the `sha256sum`  tool

Comment (piyush-garg) [reply RELEASE_PROCESS.md]: yaa both of them works

Comment (danielhelfand) [reply RELEASE_PROCESS.md]: Maybe more generally to get sha256 and say the example below uses `openssl`? I'm not sure which too is more commonly used.

File: RELEASE_PROCESS.md [hunk 3]
```diff
@@ -39,9 +39,19 @@ dnf makecache
 dnf upgrade tektoncd-cli
 ```
 
-- Make an update for [homebrew-core](https://github.com/Homebrew/homebrew-core/blob/master/Formula/tektoncd-cli.rb) for tektoncd-cli formula, and make sure you get the proper `sha256sum` and update it like this:
-
-  https://github.com/Homebrew/homebrew-core/pull/46492
+- Make an update for [homebrew-core](https://github.com/Homebrew/homebrew-core/blob/master/Formula/tektoncd-cli.rb) for tektoncd-cli formula.
+
+  * Make sure you get the proper `sha256sum` You need to download the `Source Code (tar.gz)` from the release and hit below command for the respective release
```
Comment (danielhelfand) [inline RELEASE_PROCESS.md]:
```suggestion
  * Make sure you get the proper `sha256sum`. You need to download the `Darwin_x86_64.tar.gz` from the [release section of the cli GitHub](https://github.com/tektoncd/cli/releases) and run the command below for the release version you are working on:
```

File: RELEASE_PROCESS.md [hunk 4]
```diff
@@ -39,9 +39,19 @@ dnf makecache
...
+  
+    ```shell script
+    openssl dgst --sha256 cli-0.7.1.tar.gz 
+    ```
+    
+    You will get a output like
```
Comment (danielhelfand) [inline RELEASE_PROCESS.md]:
```suggestion
    You will get similar output to what is shown below:
```

File: RELEASE_PROCESS.md [hunk 5]
```diff
@@ -39,9 +39,19 @@ dnf makecache
...
+    
+    `SHA256(cli-0.7.1.tar.gz)= 72df3075303d2b0393bae9a0f9e9b8441060b8a4db57e613ba8f1bfda03809b5`
+
+  * Raise a PR to update like [this](https://github.com/Homebrew/homebrew-core/pull/46492)
```
Comment (danielhelfand) [inline RELEASE_PROCESS.md]:
```suggestion
  * Raise a PR to update like [this](https://github.com/Homebrew/homebrew-core/pull/46492) to Homebrew Core
```

File: RELEASE_PROCESS.md (hunk 1 above)
Comment (piyush-garg) [reply RELEASE_PROCESS.md]: done

File: RELEASE_PROCESS.md (hunk 2 above)
Comment (piyush-garg) [reply RELEASE_PROCESS.md]: i have changed the command to `sha256sum`

File: RELEASE_PROCESS.md (hunk 3 above)
Comment (piyush-garg) [reply RELEASE_PROCESS.md]: But I have addressed the other changes, but it is Source Code tarball not Darwin

File: RELEASE_PROCESS.md (hunk 5 above)
Comment (piyush-garg) [reply RELEASE_PROCESS.md]: done

File: RELEASE_PROCESS.md (hunk 3 above)
Comment (danielhelfand) [reply RELEASE_PROCESS.md]: I guess it can be either, so works for me.

File: tekton/README.md [hunk 6]
```diff
@@ -32,6 +32,11 @@ commits.
 It will them use your Kubernetes cluster with Tekton and apply what needs to be done for
 running the release.
 
+For running the release pipeline, you will require a github token, which script will ask
+to provide. You can generate a GitHub token using the following 
+[instructions](https://help.github.com/en/github/authenticating-to-github/creating-a-personal-access-token-for-the-command-line). 
+Your token needs following access `admin:org, read:packages, repo, write:packages`
```
Comment (danielhelfand) [inline tekton/README.md]: Still not %100 on these permissions, but they work for me and we can iterate on this.

Comment (piyush-garg) [reply tekton/README.md]: yaa I used the same permissions and they worked for me.

File: RELEASE_PROCESS.md (hunk 3 above)
Comment (piyush-garg) [reply RELEASE_PROCESS.md]: No, Darwin has different `sha` and CI failed for that

**Pull request merged**